quake2.6
.\"                                      Hey, EMACS: -*- nroff -*-
.TH QUAKE2 6 "2016-01-16"
.SH NAME
quake2 \- classic first person shooter
.SH SYNOPSIS
.B quake2
.RI [ WRAPPER-OPTIONS ]
.RI [ ENGINE-OPTIONS ]
.RI [ ENGINE-COMMANDS ] ...
.br
.SH DESCRIPTION
This manual page documents briefly the
.B quake2
command.
.PP
\fBQuake II\fP is a popular first-person shooter game, released in 1997
by iD Software.  This wrapper script will launch the Quake II engine.  The
game data needs to be installed independently using the 'game-data-packager'
program, and by default it will be looked for at \fI/usr/share/games/quake2\fR
and \fI/usr/share/games/quake2-demo\fR.
.PP
This wrapper script accepts a few GNU-style options.
Any further arguments are passed to the engine unmodified.
Consult the engine documentation for full details, but most engines support a
syntax where you can give console commands by prefixing with a plus sign.
.PP
Additionally, if \fBquake2\fR is run via a symbolic link whose name
contains \fBreckoning\fR, \fBxatrix\fR, \fBgroundzero\fR or \fBrogue\fR,
it will play one of the Quake II mission packs, \fBThe Reckoning\fR
(by Xatrix Entertainment) or \fBGround Zero\fR (by Rogue
Entertainment). This requires that additional data files are
installed.
.SH OPTIONS
These options are accepted for \fBWRAPPER-OPTIONS\fR in the
synopsis.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-version
Show version of program.
.TP
.B \-\-engine=\fBENGINE\fR
Select a non-default Quake II engine binary
.TP
.B \-\-demo
Use the demo data, even if you have both the demo and the full version
installed
.SH EXAMPLES
Play the demo in cooperative mode:
.B quake2 --demo +map demo1 +set coop 1
.TP
Connect to a server for multiplayer:
.B quake2 +connect quake2.example.com
.TP
Run the client as a temporary dedicated server (see the \fBquake2-server\fR
package for more server infrastructure):
.B quake2 +set dedicated 1 +set deathmatch 1 +map q2dm1
.SH SEE ALSO
.BR quake2-server (6),
.BR game-data-packager (6).
.SH AUTHOR
This manual page, and the quake2 wrapper script, were written by Simon McVittie
<smcv@debian.org>, for the Debian project (and may be used by others).
They are based on the Quake packaging by David Banks and Simon McVittie.


debian/changelog
game-data-packager (44) unstable; urgency=medium

  New game support:

  [ Alexandre Detiste ]
  * Enemy Territory: Quake Wars, using the proprietary x86 binaries
  * ScummVM games:
    - Amazon: Guardians of Eden
    - Beavis and Butthead in Virtual Stupidity
    - The Bizarre Adventures of Woodruff and the Schnibble
    - Conquests of Camelot: The Search for the Grail
    - Conquests of the Longbow: The Legend of Robin Hood
    - Freddy Pharkas: Frontier Pharmacist
    - Hoyle's Official Book of Games
    - Jones in the Fast Lane
    - Leather Goddesses of Phobos 2
    - Mixed-Up Mother Goose
    - Nippon Safes Inc.
    - Once Upon A Time: Little Red Riding Hood
    - Pajama Sam series
    - Rex Nebular and the Cosmic Gender Bender
    - Ringworld 1 & 2
    - Rodney's Funscreen
    - The Lost Files of Sherlock Holmes: The Case of the Rose Tattoo
    - Simon the Sorcerer's Puzzle Pack
    - Tony Tough and the Night of Roasted Moths
    - Touché: The Adventures of the Fifth Musketeer
    - Troll's Tale
    - Urban Runner
    - Ween: The Prophecy
    - Winnie the Pooh in the Hundred Acre Wood
  * Douglas Adams's "Bureaucracy" text game
  * The Elder Scrolls III: Morrowind (Closes: #776541)
  * Ultima VII (Closes: #784733)

  [ Stephen Kitt ]
  * Infocom games:
    - A Mind Forever Voyaging
    - Ballyhoo
    - Border Zone
    - Cutthroats
    - Deadline
    - Enchanter
    - Hollywood Hijinx
    - Infidel
    - Leather Goddesses of Phobos
    - The Lurking Horror
    - Moonmist
    - Nord and Bert Couldn't Make Head or Tail of It
    - Plundered Hearts
    - Seastalker
    - Sherlock: The Riddle of the Crown Jewels
    - Sorcerer
    - Spellbreaker
    - Starcross
    - Stationfall
    - Suspect
    - Suspended
    - Trinity
    - Wishbringer
    - Witness

  [ Simon McVittie ]
  * Unreal (classic or Gold), using the proprietary x86 binaries

  Support for new languages and versions:

  [ Alexandre Detiste ]
  * Discworld 1: add de/fr/es/it floppy version
  * RTCW: add Italian version, Thanks to iortcw upstream

  [ Stephen Kitt ]
  * Cruise for a Corpse: add French CD version
  * The Dig: add English CD version

  Enhancements and bug fixes:

  [ Alexandre Detiste ]
  * fix en_GB auto-detection
  * fix autodownloading of GrimFandango Italian demo
  * fix 'gog', 'atlantis' & 'baldurs-gate-2' targets (Closes: #804804)
  * use steamcmd to download Linux & Windows game assets (Closes: #775365)
  * re-add Doom 2 - MasterLevels launcher icon that got lost in v42 or v43

  [ Simon McVittie ]
  * runtime: add a generic Gtk launcher, initially for Unreal
  * Add an experimental AppArmor profile for the non-free Unreal binaries,
    to protect the rest of the system from any exploits

  [ Stephen Kitt ]
  * Update the GOG installer for The Dig (English).
  * Add Sam & Max CD manual.

  Internal changes:

  [ Alexandre Detiste ]
  * fedora: prepare initial release
  * make install: fix inclusion of game_data_packager/unpack in .deb
  * man pages: install man pages in Makefile
  * refactor packaging for various games
  * avoid computing known md5 again in our dh_md5sums implementation
  * steam: semi-automatically tag remaining native games

  [ Simon McVittie ]
  * quake, quake2: make aliases consistently strings
  * Always build vfs.zip, and optionally use it even when uninstalled
  * to_yaml: turn sets into sorted lists, and output more fields
  * Add a test that various forms of the game data are equivalent
  * GameData: do not crash if _iter_expand_groups changes self.files
  * build: allow tar.* as an unpacker format
  * build: add support for xdelta version 1 patches
  * Add an abstraction and limited auto-detection for archive unpacking
  * Add library code to extract Unreal mods and installers
  * make-template: always capture SHA1 and SHA256: we can edit them out
    later if desired, but if the g-d-p maintainers don't own a particular
    game, more information is better than less
  * make-template: use groups
  * make-template: refactor to support auto-detection and recursion into
    (some) archives
  * make-template: preserve case for files from InnoExtract
  * make-template: preserve full name of license files
  * Add support for forcing a package's short and long description
  * Allow packages to be marked as deliberately empty
  * Generalize support for $install_to in symlinks to cover $assets, etc.
  * Allow any file to be made executable by the installation process
  * Do not try to obtain non-preferred alternatives for files
  * Accept $GDP_DEBUG as a synonym for $DEBUG
  * Switch Vcs-Git to https (see #810378)

 -- Simon McVittie <smcv@debian.org>  Fri, 22 Jan 2016 23:01:25 +0000

game-data-packager (43) unstable; urgency=medium

  New game support:

  [ Alexandre Detiste ]
  * X-COM: UFO Defense (Closes: #793017)
  * ScummVM games:
    - 3 Skulls of the Toltecs
    - Bargon Attack
    - The Black Cauldron
    - Blue Force
    - Castle of Dr. Brain & The Island of Dr. Brain
    - Chivalry is Not Dead
    - Codename: ICEMAN
    - Cruise for a Corpse
    - EcoQuest 1 & 2
    - Enclosure (fan-made game)
    - Eye of the Beholder 1 & 2
    - The Feeble Files
    - Future Wars (English version)
    - Gold Rush!
    - Hank's Quest (fan-made game)
    - Hopkins FBI
    - Hugo's House of Horrors and its two sequels
    - Lands of Lore: The Throne of Chaos
    - Laura Bow 1 & 2
    - The Lost Files of Sherlock Holmes: The Case of the Serrated Scalpel
       (support for this game is currently being added to scummvm)
    - Lost in Time
    - Manhunter 1 & 2
    - The Manhole
    - The Neverhood
    - Personal Nightmare
    - Slater & Charlie Go Camping
    - Toonstruck

  [ Markus Koschany ]
  * Wolfenstein: Enemy Territory, for the etlegacy engine (Closes: #780446)
  * Baldur's Gate - The Original Saga (GOG.com English Linux installer)
    for the gemrb engine (Closes: #776540)
  * Baldur's Gate 2 (GOG.com English Linux installer) for the gemrb engine
    (Closes: #800712)

game_data_packager/games/quake2.py
#!/usr/bin/python3
# encoding=utf-8
#
# Copyright © 2015 Simon McVittie <smcv@debian.org>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License version 2
# as published by the Free Software Foundation.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
#
# You can find the GPL license text on a Debian system under
# /usr/share/common-licenses/GPL-2.

import glob
import logging
import os
import subprocess
import tarfile

from .. import (GameData)
from ..build import (PackagingTask)

logger = logging.getLogger('game-data-packager.games.quake2')

class Quake2GameData(GameData):
    def construct_task(self, **kwargs):
        return Quake2Task(self, **kwargs)

class Quake2Task(PackagingTask):
    def fill_dest_dir(self, package, destdir):
        if not super(Quake2Task, self).fill_dest_dir(package, destdir):
            return False

        if package.name not in ('quake2-reckoning-data',
                                'quake2-groundzero-data'):
            return True

        subdir = {
            'quake2-groundzero-data': 'rogue',
            'quake2-reckoning-data': 'xatrix',
        }[package.name]

        installdir = os.path.join(destdir, 'usr', 'share', 'games', 'quake2')
        unpackdir = os.path.join(self.get_workdir(), 'tmp',
                package.name + '.build.d')

        tars = list(glob.glob(os.path.join(installdir, '*.tar.xz')))
        assert len(tars) == 1, tars
        expect_dir = os.path.basename(tars[0])
        expect_dir = expect_dir[:len(expect_dir) - 7]

        with tarfile.open(tars[0], mode='r:xz') as tar:
            tar.extractall(unpackdir)

        subprocess.check_call(['make', '-C',
            os.path.join(unpackdir, expect_dir), '-s', '-j5'])
        subprocess.check_call(['install', '-s', '-m644',
            os.path.join(unpackdir, expect_dir, 'release', 'game.so'),
            os.path.join(installdir, subdir, 'game.so')])

        return True

GAME_DATA_SUBCLASS = Quake2GameData
